docs/src/main/asciidoc/security-webauthn.adoc

```diff
@@ -0,0 +1,1300 @@
+        .get("/api/users/me")
+        .then()
+        .statusCode(200)
+        .body(Matchers.is(userName));
+        
+        // admin API?
+        if(user == User.ADMIN) {
+            RestAssured.given().filter(cookieFilter)
+            .when()
+            .get("/api/admin")
+            .then()
+            .statusCode(200)
+            .body(Matchers.is("admin"));
+        } else {
+            RestAssured.given().filter(cookieFilter)
+            .when()
+            .get("/api/admin")
+            .then()
+            .statusCode(403);
+        }
+    }
+
+    private void verifyLoggedOut(Filter cookieFilter) {
+        // public API still good
+        RestAssured.given().filter(cookieFilter)
+        .when()
+        .get("/api/public")
+        .then()
+        .statusCode(200)
+        .body(Matchers.is("public"));
+        // public API user name
+        RestAssured.given().filter(cookieFilter)
+        .when()
+        .get("/api/public/me")
+        .then()
+        .statusCode(200)
+        .body(Matchers.is("<not logged in>"));
+
+        // user API not accessible
+        RestAssured.given()
+        .filter(cookieFilter)
+        .redirects().follow(false)
+        .when()
+        .get("/api/users/me")
+        .then()
+        .statusCode(302)
+        .header("Location", Matchers.is("http://localhost:8081/"));
+        
+        // admin API not accessible
+        RestAssured.given()
+        .filter(cookieFilter)
+        .redirects().follow(false)
+        .when()
+        .get("/api/admin")
+        .then()
+        .statusCode(302)
+        .header("Location", Matchers.is("http://localhost:8081/"));
+    }
+}
+----
+
+For this test, since we're testing both the provided callback endpoint, which updates users
+in its `WebAuthnUserProvider` and the manual `LoginResource` endpoint, which deals with users
+manually, we need to override the `WebAuthnUserProvider` with one that doesn't update the
+`scooby` user:
+
+[source,java]
+----
+package org.acme.security.webauthn.test;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.acme.security.webauthn.MyWebAuthnSetup;
+
+import io.quarkus.test.Mock;
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.auth.webauthn.Authenticator;
+
+@Mock
+@ApplicationScoped
+public class TestUserProvider extends MyWebAuthnSetup {
+    @Override
+    public Uni<Void> updateOrStoreWebAuthnCredentials(Authenticator authenticator) {
+        // delegate the scooby user to the manual endpoint, because if we do it here it will be
+        // created/udated twice
+        if(authenticator.getUserName().equals("scooby"))
+            return Uni.createFrom().nullItem();
+        return super.updateOrStoreWebAuthnCredentials(authenticator);
+    }
+}
+----
+
+[[configuration-reference]]
+== Configuration Reference
+
+include::{generated-dir}/config/quarkus-security-webauthn.adoc[opts=optional, leveloffset=+1]
+
+== References
+
+* xref:security.adoc[Quarkus Security]
```

Comment:
@FroMage This is a very good guide - can you please add a micro section to `security.adoc` itself linking to this guide, above the `OIDC` section there, since `security.adoc` is like a dashboard pointing to all the security related docs.

The other thing which I believe is important to do now is to add a short note/section to this new guide clarifying that it does not replace OIDC but rather, as you mentioned the other day, introduces a password less authentication which can be used with or without OIDC as a way to address an authentication challenge.

Reply:
Good point.